secret/secret.go

```diff
@@ -0,0 +1,96 @@
+package secret
+
+import (
+	"log"
+	"os"
+
+	"github.com/SUSE/scf-secret-generator/model"
+	"github.com/SUSE/scf-secret-generator/password"
+	"github.com/SUSE/scf-secret-generator/ssh"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+)
+
+// The name of the secret stored in the kube API
+const SECRET_NAME = "secret"
+
+func GetSecrets() corev1.SecretInterface {
+	// Set up access to the kube API
+	kubeConfig, err := rest.InClusterConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	clientSet, err := kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return clientSet.CoreV1().Secrets(os.Getenv("KUBERNETES_NAMESPACE"))
```

Comment:
Is there a way to check for errors? Will `Secrets()` return `nil` if the namespace does not exist? If yes, check for it and `log.Fatal(...)` instead of letting the caller deal with it?

Reply:
No, it errors out early with a message `2017/11/21 22:25:02 namespaces "1234" not found`. I might be able to get a list of namespaces and check for it, but I don't see a lot of value in that since I'd just be checking for it and failing if it isn't found anyway.

Reply:
Yes, if it is already failing with a log message, then it is exactly doing what we want already, so it is all good.